subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskInputPropertiesIntegrationTest.groovy

```diff
@@ -400,48 +397,120 @@ apply from:'scriptPlugin.gradle'
     }
 
 
-    @NotYetImplemented
+    @Unroll
     @Issue("gradle/gradle#784")
-    def "can use a custom Serializable type from build script as input property in a never up-to-date custom Task"() {
+    def "can use a custom Serializable type from build script as input property in a custom Task"() {
         given:
         buildFile << """
             import org.gradle.api.internal.changedetection.state.InMemoryTaskArtifactCache
 
-            println "Flushing InMemoryTaskArtifactCache"
             gradle.taskGraph.whenReady {
                 gradle.services.get(InMemoryTaskArtifactCache).invalidateAll()
```

Comment:
I removed the cache invalidation. Without the fix and the cache invalidation, the test won't fail. Thanks to the invalidation, we are able to make those test fail. That's the reason why I feel uneasy with removing these lines.